examples/pentagon.ml

```diff
@@ -0,0 +1,45 @@
+open Graphics
+
+type point = { x : int; y : int }
+type pentagon = { c : point; side_length : int}
+
+type shape = Pentagon of pentagon
+
+let canvas_size = (500, 500)
+let canvas_mid = { x = fst canvas_size / 2; y = snd canvas_size / 2 }
+
+let render_shape s =
+  match s with
+  | Pentagon pent ->
+      let side = pent.side_length in
+      let x = pent.c.x in
+      let y = pent.c.y in
+      let points =
+        [|
+          (x, y + side);
+          (x - side, y);
+          (x - (side/2), y - side);
+          (x + (side/2), y - side);
+          (x + side, y);
+        |]
```

Comment:
This does not create a regular pentagon. Some sides are longer than others.

e.g. you can use a calculator to get the 5 coordinates of a regular pentagon for some constant k with this calculation:
```
1. (0, k)
2. (-k * cos18, k * sin18)
3. (-k * cos54, -k * sin54)
4. (k * cos54, -k * sin54)
5. (k * cos18, k * sin18) 
```

(54 and 18 are in degrees)

Reply:
I'm encountering the following when i try to run the file 
![image](https://github.com/Sudha247/ocaml-joy/assets/82649346/082236a3-76db-42f1-a508-b3d5b8790120)

Here is the modified code
```
open Graphics

type point = { x : int; y : int }
type pentagon = {c : point; side_length : int}

type shape = Pentagon of pentagon

let canvas_size = (500, 500)
let canvas_mid = { x = fst canvas_size / 2; y = snd canvas_size / 2 }

let render_shape s =
  match s with
  | Pentagon pent ->
    let k = pent.side_length in
    let points =
      [|
        (0, k);
        (int_of_float (-. (float_of_int k) *. cos (18. *. (Float.pi /. 180.))), int_of_float ((float_of_int k) *. sin (18. *. (Float.pi /. 180.))));
        (int_of_float (-. (float_of_int k) *. cos (54. *. (Float.pi /. 180.))), int_of_float (-. (float_of_int k) *. sin (54. *. (Float.pi /. 180.))));
        (int_of_float ((float_of_int k) *. cos (54. *. (Float.pi /. 180.))), int_of_float (-. (float_of_int k) *. sin (54. *. (Float.pi /. 180.))));
        (int_of_float ((float_of_int k) *. cos (18. *. (Float.pi /. 180.))), int_of_float ((float_of_int k) *. sin (18. *. (Float.pi /. 180.))));
      |]
    in
    draw_poly points


let pentagon ?x ?y side_length =
  let center = match (x, y) with
    | Some x, Some y -> { x; y }
    | _ -> canvas_mid in
  Pentagon { c = center; side_length }

let show shapes = List.iter render_shape shapes

let () =
  open_graph (" " ^ string_of_int (fst canvas_size) ^ "x" ^ string_of_int (snd canvas_size));
  set_color black;

  let pent = pentagon 100 in
  show [pent];

  ignore (read_line ());
  close_graph ()
```

Reply:
@joanita-51 this is too complicated for an example. I didn't want you to add all the calculations inside the file, but rather do one calculation and build an example with that.

Can you please do that? I calculated a set of 5 points with k=200, please use these in `draw_poly` to build your pentagon (on a 500x500 canvas) --

```
  (250, 450)
  (60, 312)
  (132, 88)
  (368, 88)
  (440, 312)
```